### DISCUSSION TITLE
Sending blob

### DISCUSSION THREAD

**Discussion by Aymeric-Tremoulet** [PostgREST/postgrest#1906] [Q&A] [answered]
Hi everyone,

I have been clueless on how to solve a problem i got trying to send blobs.
My PostgREST (7.0.1) server is behind Nginx.

When I try to send (using POST on the table where I store the blob) my blob, I get a 502 error.
I just wanna say I'm not trying to send 100 MB files, if it can accept files from 300 KB to 3MB it should be way enough.
So, Nginx returns 502 error saying: upstream sent too big header while reading response header from upstream
If I send my request directly to PostgREST bypassing Nginx, I can see that it went well (201 Created) but there's no return code and I get a "Out of memory" error.

What can I do here to send my blob? Any help appreciated

Thanks in advance

**Comment by steve-chavez** [comment, marked as answer]
Hey @Aymeric-Tremoulet,

> So, Nginx returns 502 error saying: upstream sent too big header while reading response header from upstream

Hm, this issue might be related to https://github.com/PostgREST/postgrest/issues/1461, which was already fixed. Could you try upgrading to the latest [v8.0.0](https://github.com/PostgREST/postgrest/releases/tag/v8.0.0)?

> If I send my request directly to PostgREST bypassing Nginx, I can see that it went well (201 Created) but there's no return code and I get a "Out of memory" error.

We have a [test](https://github.com/PostgREST/postgrest/blob/main/test/memory-tests.sh#L113) for sending a 50M json and PostgREST consumes 171M in this case. Are you [specifying columns](https://postgrest.org/en/v7.0.0/api.html#specifying-columns) when sending the blob? This should be more efficient for sending big payloads.

**Reply by Aymeric-Tremoulet** [reply]
Hi,
After spending two full days on that matter, you saved the remaining of my braincells.
Upgrading PostgREST to 8.0.0 solved the problem.
Thank you very much.
